drivers/usb/usbhub_new/pnp.c

```diff
@@ -2068,7 +2068,7 @@ USBH_RestoreDevice(IN PUSBHUB_PORT_PDO_EXTENSION PortExtension,
     ASSERT(PortExtension->PortNumber > 0);
     PortData = &HubExtension->PortData[PortExtension->PortNumber - 1];
 
-    if (PortExtension->Common.SelfDevice == PortData->DeviceObject)
+    if (PortExtension->Common.SelfDevice != PortData->DeviceObject)
```

Comment:
Can such a case occur during normal work? Shouldn't we just place an assertion here?
For me it looks like a sanity check

Reply:
Yeah I was considering an assert. Will investigate whether this can actually happen.

Reply:
I'm not exactly sure _how_ this can happen, but we do already check for this condition elsewhere:
https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/usb/usbhub_new/pnp.c;hb=6122f3f436a4551e2a85318b7e6452a6fbada2c5#l2271
`USBH_PdoRemoveDevice` checks before calling USBH_SyncDisablePort, and also explicitly uses the PDO from PortData when updating the port status.
So I'll leave this without an assert just in case.